website/docs/sdk-reference/elixir.md

```diff
@@ -0,0 +1,251 @@
+
+Use `cache_expiry_seconds` option parameter to set cache lifetime.
+```elixir
+{ConfigCat, [
+    sdk_key: "YOUR SDK KEY",
+    cache_policy: ConfigCat.CachePolicy.Lazy(cache_expiry_seconds: 300)
+]}
+```
+
+Available options:
+
+| Option Parameter             | Description                                                                                   | Default |
+| ---------------------------- | --------------------------------------------------------------------------------------------- | ------- |
+| `cache_expiry_seconds` | Cache TTL.                                                                                    | 60      |
+
+### Manual polling
+Manual polling gives you full control over when the setting values are downloaded. *ConfigCat SDK* will not update them automatically. Calling `force_refresh()` is your application's responsibility.
+
+```elixir
+ConfigCat.force_refresh();
+```
+
+> `get_value()` returns `default_value` if the cache is empty. Call `force_refresh()` to update the cache.
+```elixir
+value = ConfigCat.get_value("key", "my default value") # Returns "my default value"
+ConfigCat.force_refresh();
+value = ConfigCat.get_value("key", "my default value") # Returns "value from server"
+```
+
+### Custom cache behaviour with `cache:` option parameter.
+To be able to customize the caching layer behaviour you need to implement the following interface:
+```elixir
+defmodule ConfigCat.ConfigCache do
+  alias ConfigCat.Config
+
+  @type key :: String.t()
+  @type result :: {:ok, Config.t()} | {:error, :not_found}
+
+  @callback get(key) :: {:ok, Config.t()} | {:error, :not_found}
+  @callback set(key, Config.t()) :: :ok
+end
+
+```
+
+At the moment, this is how the current `InMemoryCache` looks like:
+```elixir
+defmodule ConfigCat.InMemoryCache do
+  use GenServer
+
+  alias ConfigCat.ConfigCache
+  @behaviour ConfigCache
+
+  @spec start_link(options()) :: GenServer.on_start()
+  def start_link(options) do
+    name =
+      options
+      |> Keyword.fetch!(:cache_key)
+      |> name_from_cache_key()
+
+    GenServer.start_link(__MODULE__, :empty, name: name)
+  end
+
+  @impl ConfigCache
+  def get(cache_key) do
+    GenServer.call(name_from_cache_key(cache_key), :get)
+  end
+
+  @impl ConfigCache
+  def set(cache_key, value) do
+    GenServer.call(name_from_cache_key(cache_key), {:set, value})
+  end
+
+  defp name_from_cache_key(cache_key) do
+    String.to_atom(cache_key)
+  end
+
+  @impl GenServer
+  def init(state) do
+    {:ok, state}
+  end
+
+  @impl GenServer
+  def handle_call(:get, _from, :empty = state) do
+    {:reply, {:error, :not_found}, state}
+  end
+
+  @impl GenServer
+  def handle_call(:get, _from, state) do
+    {:reply, {:ok, state}, state}
+  end
+
+  @impl GenServer
+  def handle_call({:set, value}, _from, _state) do
+    {:reply, :ok, value}
+  end
+end
+```
```

Comment:
I don't think we should include this here; it's not really necessary and is going to be a real pain to keep up-to-date with the code.

I might instead just mention these things:
- You must implement (either explicitly or implicitly) the `ConfigCache` behaviour
- It is the responsibility of the calling application to supervise the cache if it needs to be supervised.

If you want to include a link to the InMemoryCache implementation, that'd be OK, but probably not necessary.